force-app/main/default/lwc/geGift/geSoftCredits.js

```diff
@@ -35,6 +38,9 @@ class SoftCredits {
     }
 
     addProcessedSoftCredits(processedSoftCredits) {
+        if (typeof processedSoftCredits === 'string') {
+            processedSoftCredits = JSON.parse(processedSoftCredits);
+        }
         this._processedSoftCredits = [ ...processedSoftCredits ];
```

Comment:
I don't think we need to use the spread operator here, this could be written as `this._processedSoftCredits = processedSoftCredits;`